Core/src/org/sleuthkit/autopsy/datamodel/DhsImageCategory.java

```diff
@@ -124,13 +124,25 @@ public String toString() {
 
     synchronized public Node getGraphic() {
         if (snapshot == null) {
+            snapshot = generateSnapshot();
+        }
+        return new ImageView(snapshot);
+    }
+
+    private Image generateSnapshot() {
+        if (Platform.isFxApplicationThread()) {
+            //We generate the 'icons' for the categories by taking a snapshot of ssimple custom Node.  This must be done on the JFX thread.
             Region region = new Region();
             region.setBackground(new Background(new BackgroundFill(getColor(), CORNER_RADII_4, Insets.EMPTY)));
             region.setPrefSize(16, 16);
             region.setBorder(new Border(new BorderStroke(getColor().darker(), BorderStrokeStyle.SOLID, CORNER_RADII_4, BORDER_WIDTHS_2)));
             Scene scene = new Scene(region, 16, 16, Color.TRANSPARENT);
-            snapshot = region.snapshot(null, null);
+            return region.snapshot(null, null);
+        } else {
+            //if we are not already on the JFX thread, do the generation on the javafx thread and block until done.
+            SettableFuture<Image> imageFuture = SettableFuture.create();
+            Platform.runLater(() -> generateSnapshot());
+            return Futures.getUnchecked(imageFuture); //block until snapshot is generated. It should be quick.
```

Comment:
Quite likely I'm missing something here, but doesn't imageFuture need to be set somewhere for it to unblock from here?

Reply:
gah, of course.  I had that, and then i was cleaning up and I must have missed it.